core/spec/services/spree/cart/remove_line_item_spec.rb

```diff
@@ -16,11 +16,30 @@ module Spree
         expect { execute }.to change { order.line_items.count }.by(-1)
         expect(execute).to be_success
         expect(value).to eq line_item
-        order.reload
         expect(order.amount).to eq 0
       end
     end
 
+    context 'remove line item' do
+      let(:variant1) { create :variant, price: 20 }
+      let(:variant2) { create :variant, price: 10 }
+      let(:line_item1) { create :line_item, variant: variant1, price: nil }
+      let(:line_item2) { create :line_item, variant: variant2, price: nil }
+      let(:order) { create :order, line_items: [line_item1, line_item2] }
```

Comment:
All this is unnecessary. There is already a factory that cna create an order with line items. All you need to do is:
```ruby
let(:order) { create(:order_with_line_items, line_items_count: 2) }
```